### DISCUSSION TITLE
Document benefits of Distant.nvim over vanilla SSH

### DISCUSSION THREAD

**Discussion by edmondop** [chipsenkbeil/distant.nvim#140] [General]
Because Neovim is a terminal-based editor and not a GUI, it is possible to `ssh` into the remote machines, checkout the dotfiles and just launch Neovim. If you use packer or lazy, they will simply install the plugins in your dotfiles at startup.

However, there are reasons/use cases for which Distant could be preferred (i.e. maybe the distant protocol works better than plain ssh with slow ping?), or when you want to customize the setup depending on the remote, etc.

**Comment by chipsenkbeil** [comment]
@edmondop moving this to a discussion since it's more in line with this space. 😄 

So far, there's an FAQ discussing the advantages of this plugin over neovim's remote TUI. It also discusses pros and cons versus sshfs.

https://distant.dev/editors/neovim/faq/

The reasons that matter to me are:

1. Latency - I can type within neovim on my local machine without having stutter. When I worked to and from work on a bus, my connection was spotty and also resulted in a lot of lag in keypresses.
2. Resiliency - when using the distant server (not the ssh client that this plugin also supports), the plugin can reconnect to your server seamlessly. Citing the bus again, this was helpful for me to not be interrupted when my connection was lost. It's also nice when moving around the office building.
3. Locality - I can have my plugins installed on my laptop and not worry about having to re-install them on a remote machine. For work, we have a lot of rules regarding what can be installed on the remote machine, and how & when you can connect to the outside world. It's easier for me to support copying over a single binary, or using the ssh client mode, rather than setting up the remote machine with neovim and all of my plugins.

**Reply by edmondop** [reply]
Thanks for your clarification. Do you know remote-nvim plugin? It seems to have similar goals and I don't fully understand the strengths and benefits. It doesn't rely on Distant

**Reply by chipsenkbeil** [reply]
Distant offers more features, to my knowledge. Reconnecting when losing a connection, running a remote shell, and running language servers on the remote machine are three I use.

Distant also offers file search via ripgrep api, which was implemented into a telescope integration.

But I'll also say that I haven't maintained distant very much over the last year. If you want more support and activity, I'd recommend another plugin. :)

I'm glad to see more plugins popping up to address this space!

**Reply by purepani** [reply]
Is this plugin/cli tool going to be something you want to keep developing then?

**Reply by chipsenkbeil** [reply]
> Is this plugin/cli tool going to be something you want to keep developing then?

Yes, it's still maintained and being developed; however, I manage a variety of other software projects so the speed at which I'm rolling out updates is pretty slow. I'm always open for new contributions, though!
